web/src/compose_notifications.ts

```diff
@@ -245,6 +248,71 @@ export function notify_messages_outside_current_search(messages: Message[]): voi
     }
 }
 
+export function maybe_show_one_time_messages_fading_banner(): void {
+    const is_faded = $(".recipient_row").hasClass("message-fade");
+
+    if (is_faded) {
```

Comment:
Use the early return pattern to avoid indenting so much code.

We can also easily adjust the selector to check if any of the faded messages are actually in the viewport?

Reply:
Done.